src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,467 @@
+        pool.withdraw(total);
+        daiJoin.join(address(pool), total);
+
+        // normalized debt == erc20 DAI to pool (Vat rate for this ilk fixed to 1 RAY)
+
+        if (mode == Mode.NORMAL) {
+            vat.frob(ilk, address(pool), address(pool), address(pool), -int256(amount), -int256(amount));
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(pool), vow, _mul(fees, RAY));
+        } else if (mode == Mode.MODULE_CULLED) {
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(pool), vow, _mul(total, RAY));
+        } else {
+            // This can be done with the assumption that the price of 1 aDai equals 1 DAI.
+            // That way we know that the prev End.skim call kept its gap[ilk] emptied as the CDP was always collateralized.
+            // Otherwise we couldn't just simply take away the collateral from the End module as the next line will be doing.
+            vat.slip(ilk, address(end), -int256(amount));
+            vat.move(address(pool), vow, _mul(total, RAY));
+        }
+
+        emit Unwind(ilk, amount);
+    }
+
+    function exec(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        uint256 availableAssets = pool.maxWithdraw();
+
+        if (vat.live() == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED
+            );
+        } else if (live == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // This module caged
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL
+            );
+        } else {
+            // Normal path
+            (uint256 totalAssets, uint256 targetAssets) = pool.calcSupplies(availableAssets);
+
+            if (targetAssets > totalAssets) {
+                _wind(ilk_, pool, targetAssets - totalAssets);
+            } else if (targetAssets < totalAssets) {
+                _unwind(
+                    ilk_,
+                    pool,
+                    totalAssets - targetAssets,
+                    availableAssets,
+                    Mode.NORMAL
+                );
+            }
+        }
+    }
+
+    // --- Collect Interest ---
+    function reap(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(live == 1, "DssDirectDepositHub/no-reap-during-cage");
```

Comment:
@hexonaut do you think this should be a live check on:

1. Hub
2. Pool
3. Hub and Pool
I'm leaning towards 3. lmk what you think

Reply:
So if we keep the global cage why does `reap` need to be blocked? It's a fairly harmless function that just collects interest. I don't think this needs to be blocked in the case either the ilk or hub are caged.